examples/plan98-client/was-hello.js

```diff
@@ -0,0 +1,254 @@
+
+  const { host } = $.learn()
+  const credentials = localStorage.getItem('was/signer')
+
+  if (credentials) {
+    signer = await Ed25519Signer.fromJSON(credentials)
+  } else {
+    // This signer can create cryptographic signatures
+    signer = await Ed25519Signer.generate()
+    localStorage.setItem('was/signer', JSON.stringify(signer.toJSON()))
+  }
+  const storageId = host
+  const storageUrl = new URL(storageId)
+  const storage = new StorageClient(storageUrl)
+
+  // create the space with signer so all requests get signed by it
+  const space = storage.space({
+    signer,
+    id: `urn:uuid:${target.id}`
+  })
+
+  const resource = space.resource(target.getAttribute('src') || '/tmp')
+  const response = await resource.get()
+    .then(res => {
+      if (res.status === 200) {
+        const indexUrl = new URL(resource.path, storageUrl)
+        $.teach({ home: indexUrl.toString() })
+      }
+      return res
+    })
+    .catch(e => {
+      console.debug(e)
+      toast(e.message, { type: 'error' })
+    })
+
+  $.teach({ loading: false })
+}
+
+async function publish(spaceId, src) {
+  const { host } = $.learn()
+  const storageId = host
+  const storageUrl = new URL(storageId)
+  const storage = new StorageClient(storageUrl)
+
+  // create the space with signer so all requests get signed by it
+  const space = storage.space({
+    signer,
+    id: `urn:uuid:${spaceId}`
+  })
+
+  const linkset = space.resource(`linkset`)
+  const spaceObject = {
+    controller: signer.controller,
+    // configure which resource to use as a linkset
+    link: linkset.path,
+  }
+  const spaceObjectBlob = new Blob(
+    [JSON.stringify(spaceObject)],
+    { type: 'application/json' },
+  )
+
+  // send PUT request to update the space
+  const responseToPutSpace = await space.put(spaceObjectBlob)
+    .then(res => {
+      console.debug({ res })
+      toast(JSON.stringify(res), { type: 'success' })
+      return res
+    })
+    .catch(e => {
+      console.debug(e)
+      toast(e.message, { type: 'error' })
+    })
+
+  if (!responseToPutSpace.ok) throw new Error(
+    `Failed to put space: ${responseToPutSpace.status} ${responseToPutSpace.statusText}`, {
+    cause: {
+      responseToPutSpace
+    }
+  })
+  if (!responseToPutSpace) return
+
+  // GET the space to make sure the PUT persisted it
+  const responseToGetSpace = await space.get()
+    .then(res => {
+      console.debug({ res })
+      toast(JSON.stringify(res), { type: 'success' })
+      return res
+    })
+    .catch(e => {
+      console.debug(e)
+      toast(e.message, { type: 'error' })
+    })
+
+  if (!responseToGetSpace.ok) throw new Error(
+    `Failed to get space: ${responseToGetSpace.status} ${responseToGetSpace.statusText}`, {
+    cause: {
+      responseToGetSpace
+    }
+  })
+  if (!responseToGetSpace) return
+
+  const index = space.resource(src)
+  const blobForIndex = new Blob(['<!doctype html><h1>Hello WAS!</h1>'], { type: 'text/html' })
```

Comment:
@tylerchilds It still feels like this is showing something but not necessarily plan98. Are there any plan98 web components we can put in this HTML so that once it is loading it is doing plan98 pretty looking stuff?